src/moneypenny/handlers/external.py

```diff
@@ -0,0 +1,150 @@
+"""Handlers for the app's external root, ``/moneypenny/``."""
+
+from fastapi import (
+    APIRouter,
+    BackgroundTasks,
+    Depends,
+    HTTPException,
+    Response,
+)
+from fastapi.responses import PlainTextResponse
+from safir.metadata import get_metadata
+
+from ..config import config
+from ..dependencies import moneypenny_dependency
+from ..exceptions import K8sApiException, PodNotFound
+from ..models import Dossier, Index
+from ..moneypenny import Moneypenny
+
+__all__ = ["get_index", "external_router"]
+
+external_router = APIRouter()
+"""FastAPI router for all external handlers."""
+
+
+@external_router.get(
+    "/",
+    description=("Returns metadata about Moneypenny."),
+    response_model=Index,
+    response_model_exclude_none=True,
+    summary="Application metadata",
+)
+async def get_index(
+    moneypenny: Moneypenny = Depends(moneypenny_dependency),
+) -> Index:
+    """GET ``/moneypenny/`` (the app's external root).
+
+    Customize this handler to return whatever the top-level resource of your
+    application should return. For example, consider listing key API URLs.
+    When doing so, also change or customize the response model in
+    `moneypenny.models.Index`.
+
+    By convention, the root of the external API includes a field called
+    ``metadata`` that provides the same Safir-generated metadata as the
+    internal root endpoint.
+    """
+    metadata = get_metadata(
+        package_name="moneypenny",
+        application_name=config.name,
+    )
+    return Index(quip=moneypenny.quip().strip(), metadata=metadata)
+
+
+@external_router.get(
+    "/{username}",
+    description=(
+        "Because Moneypenny tidies up rather quickly, 200 and 202 are"
+        " unlikely and 404 is common after successful execution."
+    ),
+    responses={
+        200: {"description": "Pod for this user has run successfully"},
+        202: {"description": "Pod for this user is currently running"},
```

Comment:
I gather this isn't something to change, but I didn't really think to use a 202 in a GET context instead of putting status in the body.

Reply:
Yeah, I'm not a big fan of the API here. I left it alone for this round of updates, but I think in the future we should rethink how we want the API to work. Given how we use it, I wonder if just blocking on the pod completing would be a better approach.